rets/http/parsers/parse.py

```diff
@@ -16,7 +16,18 @@
 
 def parse_xml(response: ResponseLike) -> etree.Element:
     encoding = response.encoding or DEFAULT_ENCODING
-    root = etree.fromstring(response.content.decode(encoding), parser=etree.XMLParser(recover=True))
+    try:
+        root = etree.fromstring(response.content.decode(encoding), parser=etree.XMLParser(recover=True))
+    except ValueError as e:
+        if str(e) == "Unicode strings with encoding declaration are not supported. Please use bytes input or XML fragments without declaration.":
+            try:
+                # try to parse bytes directly, rather than from string
+                root = etree.XML(response.content)
+            except Exception:
+                # unable to parse bytes. abort, and re-raise original error
+                raise e
+        else:
+            raise e
```

Comment:
Solution taken from: https://stackoverflow.com/a/38244227

### Details of the issue:

`response.content` = `b'<?xml version=\'1.0\' encoding=\'US-ASCII\'?>\n<RETS ReplyCode="20201" ReplyText="No records found"/>\n'`
`response.encoding` = `'ISO-8859-1'`


Stack trace:
```  File ".../python3.7/site-packages/rets/client/resource_class.py", line 61, in search
    **kwargs,
  File ".../python3.7/site-packages/rets/http/client.py", line 245, in search
    raise e
  File ".../python3.7/site-packages/rets/http/client.py", line 239, in search
    parse_search(response)
  File ".../python3.7/site-packages/rets/http/parsers/parse.py", line 126, in parse_search
    elem = parse_xml(response)
  File ".../python3.7/site-packages/rets/http/parsers/parse.py", line 19, in parse_xml
    root = etree.fromstring(response.content.decode(encoding), parser=etree.XMLParser(recover=True))
  File "src/lxml/etree.pyx", line 3237, in lxml.etree.fromstring
  File "src/lxml/parser.pxi", line 1891, in lxml.etree._parseMemoryDocument
ValueError: Unicode strings with encoding declaration are not supported. Please use bytes input or XML fragments without declaration.
```